src/operations_problem_templates.jl

```diff
@@ -0,0 +1,135 @@
+struct EconomicDispatchProblem <: PowerSimulationsOperationsProblem end
+struct UnitCommitmentProblem <: PowerSimulationsOperationsProblem end
+struct AGCReserveDeployment <: PowerSimulationsOperationsProblem end
+
+function _default_devices_uc()
+    return [
+        DeviceModel(PSY.ThermalStandard, ThermalBasicUnitCommitment),
+        DeviceModel(PSY.RenewableDispatch, RenewableFullDispatch),
+        DeviceModel(PSY.RenewableFix, FixedOutput),
+        DeviceModel(PSY.HydroEnergyReservoir, HydroDispatchRunOfRiver),
+        DeviceModel(PSY.HydroDispatch, HydroDispatchRunOfRiver),
+        DeviceModel(PSY.PowerLoad, StaticPowerLoad),
+        DeviceModel(PSY.InterruptibleLoad, InterruptiblePowerLoad),
+        DeviceModel(PSY.Line, StaticBranch),
+        DeviceModel(PSY.Transformer2W, StaticBranch),
+        DeviceModel(PSY.TapTransformer, StaticBranch),
+        DeviceModel(PSY.HVDCLine, HVDCDispatch),
+    ]
+end
+
+function _default_devices_dispatch()
+    default = _default_devices_uc()
+    default[1] = DeviceModel(PSY.ThermalStandard, ThermalDispatch)
+    return default
+end
+
+function _default_services()
+    return [
+        ServiceModel(PSY.VariableReserve{PSY.ReserveUp}, RangeReserve),
+        ServiceModel(PSY.VariableReserve{PSY.ReserveDown}, RangeReserve),
+    ]
+end
+
+"""
+    template_unit_commitment(; kwargs...)
+
+Creates an `OperationsProblemTemplate` with default DeviceModels for a Unit Commitment
+problem.
+
+# Example
+```julia
+template = template_unit_commitment()
+```
+
+# Accepted Key Words
+- `network::Type{<:PM.AbstractPowerModel}` : override default network model settings
+- `devices::Vector{DeviceModel}` : override default `DeviceModel` settings
+- `services::Vector{ServiceModel}` : override default `ServiceModel` settings
+"""
+function template_unit_commitment(; kwargs...)
+    network = get(kwargs, :network, CopperPlatePowerModel)
+    template = OperationsProblemTemplate(network)
+    for model in get(kwargs, :devices, _default_devices_uc())
+        set_device_model!(template, model)
+    end
+
+    for model in get(kwargs, :services, _default_services())
+        set_service_model!(template, model)
+    end
+    return template
+end
+
+"""
+    template_economic_dispatch(; kwargs...)
+
+Creates an `OperationsProblemTemplate` with default DeviceModels for an Economic Dispatch
+problem.
+
+# Example
+```julia
+template = template_economic_dispatch()
+```
+
+# Accepted Key Words
+- `network::Type{<:PM.AbstractPowerModel}` : override default network model settings
+- `devices::Vector{DeviceModel}` : override default `DeviceModel` settings
+- `services::Vector{ServiceModel}` : override default `ServiceModel` settings
+"""
+function template_economic_dispatch(; kwargs...)
```

Comment:
It might be more appropriate to explicitly list the kwargs as optional parameters. As it is a user could pass misspelled arguments and you will not use them or throw errors. Do you want this function to ignore extra arguments? It's fine to use `kwargs...` to generically forward parameters. However, I think it's generally better to have some function in the stack (probably at the bottom) explicitly list them so that you catch errors.

Same comment applies to many other locations.

Reply:
agreed, I haven't totally cleaned the code in this file.